exec/java-exec/src/main/java/org/apache/drill/exec/planner/FileSystemPartitionDescriptor.java

```diff
@@ -180,7 +180,7 @@ protected void createPartitionSublists() {
 
     // build a list of DFSDirPartitionLocation.
     for (final List<String> dirs : dirToFileMap.keySet()) {
-      locations.add( new DFSDirPartitionLocation((String [])dirs.toArray(), dirToFileMap.get(dirs)));
+      locations.add( new DFSDirPartitionLocation(dirs.toArray(new String[dirs.size()]), dirToFileMap.get(dirs)));
```

Comment:
Why is this change necessary? As I already mentioned, I don't see why is it necessary to go back and forth between an array and a list in this method, so I would prefer not to introduce an unrelated change into the PR and handle it in a separate PR/JIRA.

Reply:
Test fails, when Drill builds on JDK10. Reason: Arrays.ArrayList.toArray() returns Object[] instead of String[] and cast not work.